helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -555,7 +563,7 @@ private void subscribeDataChange(String path, NotificationContext.Type callbackT
         logger.debug(_manager.getInstanceName() + " subscribe data-change. path: " + path
             + ", listener: " + _listener);
       }
-      _zkClient.subscribeDataChanges(path, this);
+      _zkClient.subscribeDataChanges(path, this, callbackType != Type.INIT);
```

Comment:
Same here, we are not checking the result.

Reply:
If CallbackType is CALLBACK, ignore it is just want we want. That is exactly the case of async removal of a task current state node from server would do. 

If CallbackType is INIT, it would not fail.